### PR TITLE
EOS-22314: SupportBundle - NorthBound interfaces

### PR DESCRIPTION
**Pull request by vastradparayya**
Signed-off-by: Parayya Vastrad <parayya.vastrad@seagate.com>

# Problem Statement
- SupportBundle - NorthBound interfaces

# Design
-  JIRA - https://jts.seagate.com/browse/EOS-22314
-  https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/316735509/Support+Bundle+Framework#North-Bound-interface-(for-CLI%2FGUI)

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: Y
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: Y
- [x] Confirm Testing was performed with installed RPM [Y/N]:  Y

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: N
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/support_framework/bundle.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Bundle:
```
Comment (sachinpunadikar) [inline py-utils/src/utils/support_framework/bundle.py:1]: Add copyright text

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+from cortx.utils.support_framework.services import ProvisionerServices
+from cortx.utils.schema import database
+from cortx.utils.cli_framework.command import Command
+
+
+class SupportBundle:
+
+    """ This Class initializes the Support Bundle Generation for CORTX. """
+    @staticmethod
+    async def get_active_nodes():
```
Comment (sachinpunadikar) [inline py-utils/src/utils/support_framework/support_bundle.py:38]: make it internal method _get_active_nodes()

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                                rc=errno.ENOENT)
+
+        symlink_path = const.SYMLINK_PATH
+        from cortx.utils.support_framework import Bundle
+        bundle_obj = Bundle(bundle_id=bundle_id, bundle_path=symlink_path,
+                            comment=comment)
+        return bundle_obj
+
+    @staticmethod
+    async def get_bundle_status(command):
```
Comment (sachinpunadikar) [inline py-utils/src/utils/support_framework/support_bundle.py:128]: rename method to get_status

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
Work on the comments

File: py-utils/src/utils/support_framework/bundle_generate.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+import shutil
+from typing import List
+from cortx.utils.schema.payload import Yaml, Tar
+from cortx.utils.support_framework import const
+from datetime import datetime
+from cortx.utils.process import SimpleProcess
+from cortx.utils.conf_store.conf_store import Conf
+from cortx.utils.log import Log
+
+ERROR = "error"
```
Comment (selvakumaar5496) [inline py-utils/src/utils/support_framework/bundle_generate.py:29]: single quote

File: py-utils/src/utils/support_framework/bundle_generate.py [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    @staticmethod
+    def _publish_log(msg, level, bundle_id, node_name, comment):
+        """
+        Format and Publish Log to ElasticSearch via Rsyslog.
+        :param msg: Message to Be added :type: str.
+        :param bundle_id: Unique Bundle Id for the Bundle :type:str.
+        :param level: Level for the Log. :type: Log.ERROR/LOG.INFO.
+        :param node_name: Name of the Node where this is running :type:str.
+        :param comment: Comment Added by user to Generate the Bundle :type:str.
+        :return: None.
```
Comment (selvakumaar5496) [inline py-utils/src/utils/support_framework/bundle_generate.py:45]: Should look like a table
For ex:
msg:            Message to Be added :type: str.
bundle_id:   Unique Bundle Id for the Bundle :type:str.
level:            Level for the Log. :type: Log.ERROR/LOG.INFO.

File: py-utils/src/utils/support_framework/bundle_generate.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+        """
+        result = "Success"
```
Comment (selvakumaar5496) [inline py-utils/src/utils/support_framework/bundle_generate.py:47]: Single quote

File: py-utils/src/utils/support_framework/bundle_generate.py [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+        # Read Commands.Yaml and Check's If It Exists.
+        cmd_setup_file = os.path.join(Conf.get("cortx_config", "install_path"),
+                                "cortx/utils/conf/support.yaml")
+        support_bundle_config = Yaml(cmd_setup_file).load()
+        if not support_bundle_config:
+            ComponentsBundle._publish_log(f"No such file {cmd_setup_file}",
+                                         ERROR, bundle_id, node_name, comment)
+            return None
+        # Path Location for creating Support Bundle.
+        path = os.path.join(Conf.get("cortx_config","support>support_bundle_path"))
```
Comment (selvakumaar5496) [inline py-utils/src/utils/support_framework/bundle_generate.py:137]: space after comma

**Review by selvakumaar5496 — CHANGES_REQUESTED** [review]
Change to single quote wherever applicable. I could see double quotes in many places for single word

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 8]
```diff
@@ -0,0 +1,225 @@
+# CORTX-CSM: CORTX Management web and CLI interface.
```
Comment (sachinpunadikar) [inline py-utils/src/utils/support_framework/support_bundle.py:1]: Correct this line  .. should be cortx-py-utils

File: py-utils/src/utils/support_framework/bundle_generate.py (hunk 7 above)
Comment (vastradparayya) [reply py-utils/src/utils/support_framework/bundle_generate.py]: Done

File: py-utils/src/utils/support_framework/bundle_generate.py (hunk 6 above)
Comment (vastradparayya) [reply py-utils/src/utils/support_framework/bundle_generate.py]: Done

File: py-utils/src/utils/support_framework/bundle_generate.py (hunk 5 above)
Comment (vastradparayya) [reply py-utils/src/utils/support_framework/bundle_generate.py]: Done

File: py-utils/src/utils/support_framework/support_bundle.py (hunk 3 above)
Comment (vastradparayya) [reply py-utils/src/utils/support_framework/support_bundle.py]: Done

File: py-utils/src/utils/support_framework/support_bundle.py (hunk 8 above)
Comment (vastradparayya) [reply py-utils/src/utils/support_framework/support_bundle.py]: Done

**Review by sachinpunadikar — APPROVED** [review]
Looks good to me

File: py-utils/src/utils/support_framework/const.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+SUPPORT_BUNDLE_TAG = 'support_bundle;'
+SUPPORT_BUNDLE = 'SUPPORT_BUNDLE'
+SOS_COMP = 'os'
+SB_COMPONENTS = 'components'
+SB_COMMENT = "comment"
```
Comment (selvakumaar5496) [inline py-utils/src/utils/support_framework/const.py:20]: please take care of this double quote

File: py-utils/src/utils/support_framework/bundle_generate.py [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import os
+import threading
+import shutil
+from typing import List
+from cortx.utils.schema.payload import Yaml, Tar
+from cortx.utils.support_framework import const
+from datetime import datetime
```
Comment (selvakumaar5496) [inline py-utils/src/utils/support_framework/bundle_generate.py:24]: As per the guideline, this should be imported before local packages right?

**Review by selvakumaar5496 — COMMENTED** [review]
Docstrings are inconsistent. Added spaces in some places. And no spaces in some places

**Review by selvakumaar5496 — APPROVED** [review]

**Comment by sachinpunadikar** [comment]
test this please

**Pull request merged**